metadata/db.go

```diff
@@ -0,0 +1,30 @@
+package metadata
+
+import (
+	"sync"
+
+	"github.com/boltdb/bolt"
+)
+
+type DB struct {
+	db *bolt.DB
+
+	storeL sync.Mutex
+	ss     map[string]*snapshotter
```

Comment:
could this data get stale or out of sync?

Reply:
Right now no, but the model is wrong and I am fixing it right now. Afterwards services will only need to load the metadata plugin and will get the snapshotter and content store from that single instance. That will prevent the dynamic registration model that it is using today, even though the registration only occurs during initialization.